### PR TITLE
Fix MSYS build

### PR DESCRIPTION
**Pull request by EssGeeEich**
This fixes the indentation in miniz.cpp, a missing cstdio include and many incorrect _WIN32 checks.
Please note that "make install" still doesn't seem to work properly.

### CODE REVIEW

**Comment by janisozaur** [comment]
Indentation fixes in `miniz.c` are taken care of in #17, it has newer version of miniz too. You may want to check if it solves your MSYS issues as well.

**Comment by EssGeeEich** [comment]
#17 fixes the miniz.c issues, but the other issues remain (The CMakeLists uses MSVC flags on GCC for Windows)

File: src/misc/window_win3264.cpp [hunk 1]
```diff
@@ -199,7 +199,7 @@ LRESULT CALLBACK Anvil::WindowWin3264::msg_callback_pfn_proc(HWND   window_handl
 void Anvil::WindowWin3264::run()
 {
     int done   = 0;
-    int result = 0;
+    //int result = 0;
```
Comment (DominikWitczakAMD) [inline src/misc/window_win3264.cpp]: Good catch. Please consider removing this variable declaration completely (and the one you commented out below as well).

**Comment by DominikWitczakAMD** [comment]
This change now features conflicts. Please resolve them.

File: src/misc/window_win3264.cpp (hunk 1 above)
Comment (EssGeeEich) [inline src/misc/window_win3264.cpp]: Thank GCC, which spits a warning for written-but-unread variables.

**Comment by EssGeeEich** [comment]
Conflicts should be resolved.

**Comment by DominikWitczakAMD** [comment]
Looks good! Thank you for your contribution, it's much appreciated.

**Pull request merged**